Docs: document TLS configuration for generic OAuth (#27006)
grafana/grafana: docs/sources/auth/generic-oauth.md

api_url =
allowed_domains = mycompany.com mycompany.org
allow_sign_up = true
tls_skip_verify_insecure = false
tls_client_cert =
tls_client_key =
tls_client_ca =
```

Set `api_url` to the resource that returns [OpenID UserInfo](https://connect2id.com/products/server/docs/api/userinfo) compatible information.

> You can also specify the TLS configuration used by the client. Set `tls_client_cert` to the path of the certificate, `tls_client_key` to the path containing the key and `tls_client_ca` to the path containing a trusted certificate authority list. 

Reviewer: Didn't we agree at some point to use the term SSL/TLS, since they're synonymous and SSL is the more popular term these days?

Author: ok, I hardly remember having this discussion.
do you suggest to change the documentation or the settings as well?
If the latter, changing the settings would be breaking change; it would require users to update their configurations. Anyway I would suggest updating the documentation fo now to be compliant with the existing implementation and decide about the configuration names later.

Reviewer: @papagian See discussion [here](https://github.com/grafana/grafana/pull/21341#discussion_r364120835). I think we've decided to standardize on "SSL", but Diana wanted us to include TLS as an alternate term so people can recognize it.

> You can also specify the SSL/TLS configuration used by the client. Set `tls_client_cert` to the path of the certificate, `tls_client_key` to the path containing the key and `tls_client_ca` to the path containing a trusted certificate authority list. 

Reviewer 2: ```suggestion
You can also specify the SSL/TLS configuration used by the client. 
- Set `tls_client_cert` to the path of the certificate. 
- Set `tls_client_key` to the path containing the key.
- Set `tls_client_ca` to the path containing a trusted certificate authority list.

```

> Finally, `tls_skip_verify_insecure` controls whether a client verifies the server's certificate chain and host name. If it is true, SSL/TLS accepts any certificate presented by the server and any host name in that certificate. In this mode, SSL/TLS is susceptible to man-in-the-middle attacks. This should be used only for testing.

Reviewer 2: ```suggestion
`tls_skip_verify_insecure` controls whether a client verifies the server's certificate chain and host name. If it is true, then SSL/TLS accepts any certificate presented by the server and any host name in that certificate. _You should only use this for testing_, because this mode leaves SSL/TLS susceptible to man-in-the-middle attacks.
```
